cunumeric/deferred.py

```diff
@@ -1132,6 +1132,40 @@ def dot(self, src1, src2, stacklevel=0, callsite=None):
                 f"dot between {rhs1_array.ndim}d and {rhs2_array.ndim}d arrays"
             )
 
+    # Create array from input array and indices
+    @profile
+    @auto_convert([1])
```

Comment:
`@auto_convert` will not work for this method. What this decorator does is that it automatically calls `to_deferred_array` on each argument at that index (in this case index `1`). However, this method gets a non-constant number of array arguments in a tuple (`choices`), so you have to call `to_deferred_array` manually on each of them.

Reply:
I see, OK!